components/script/dom/htmliframeelement.rs

```diff
@@ -228,6 +233,31 @@ impl HTMLIFrameElement {
     /// <https://html.spec.whatwg.org/multipage/#process-the-iframe-attributes>
     fn process_the_iframe_attributes(&self, mode: ProcessingMode) {
         // TODO: srcdoc
+        // https://html.spec.whatwg.org/multipage/#process-the-iframe-attributes
+        if self
+            .upcast::<Element>()
+            .has_attribute(&local_name!("srcdoc"))
+        {
+            let url = ServoUrl::parse("about:srcdoc").unwrap();
+            let document = document_from_node(self);
+            let window = window_from_node(self);
+            //added pipeline_id but still tests are failing
+            let pipeline_id = Some(window.upcast::<GlobalScope>().pipeline_id());
+            let mut load_data = LoadData::new(
+                LoadOrigin::Script(document.origin().immutable().clone()),
+                url,
+                pipeline_id,
+                Some(Referrer::ReferrerUrl(document.url())),
+                document.get_referrer_policy(),
+            );
+            load_data.srcdoc = self.get_srcdoc();
+            self.navigate_or_reload_child_browsing_context(
+                load_data,
+                NavigationType::Regular,
```

Comment:
Ah, here's the missing piece - in `navigate_or_reload_child_browsing_context` there's a special branch for NavigationType::InitialAboutBlank which ends up calling ScriptThread::handle_new_layout. If you pass that instead of NavigationType::Regular you should end up with more meaningful test results.

Reply:
The reason this matters is that Regular causes us to go to the network thread and try to make a request for about:srcdoc, which is then not a recognized URL so we get an error page. To quickly verify if this is working correctly or not, use this test page:
```html
<iframe srcdoc="<div>hi!</div><script>foo=5;</script>" onload="console.log(this.contentWindow.foo)"></iframe>
```
In the current branch, I see an iframe containing an error message. With the change I suggested, there should be an iframe with "hi!" and the terminal should show `5`.

Reply:
I have done the change. But not tested yet.

Reply:
Done. I can see the iframe and terminal shows 5. But main tests are failing. The command used: 
`/mach test-wpt tests/wpt/web-platform-tests/html/semantics/embedded-content/the-iframe-element/srcdoc_process_attributes.html`